### PR TITLE
Pincode Set Fix

### PR DESCRIPTION
**Pull request by MuckT**
### Description
A follow up to #2593 and #2608. Fixes the incorrect behavior where the wrong error message was served on pins that didn't match. This change moves the `isVerifying` navigation param, which was always passed as false, into the component state.

### Other changes
- Update mobile tests in https://github.com/valora-inc/wallet/commit/5a596f9406f51b31c484dc6e63c3530e88c23539
- Use `Buffer.from` instead of `new Buffer` - [Porting to the Buffer.from()/Buffer.alloc() API](https://nodejs.org/en/docs/guides/buffer-constructor-deprecation/#overview)
- Added e2e test for pincode reentry in https://github.com/valora-inc/wallet/pull/2629/commits/b64a82a9e1347167198b7eddaed8ef695458deda

### Tested
Tested locally on iOS

### How others should test
Perquisites: When creating a pin or changing a pin

1. Enter the first pin.
2. Enter a non-matching second pin.
3. Observe error message should be 'The PINs didn't match'
4. Should be redirected to the first step if second pin doesn't match.

### Related issues
N/A

### Backwards compatibility
Yes

### CODE REVIEW

**Review by kathaypacific — APPROVED** [review]
🚀 Looks great!!

File: e2e/src/usecases/PINChange.js [hunk 1]
```diff
@@ -24,6 +24,15 @@ export default ChangePIN = () => {
     // Then we enter the new PIN
     await enterPinUi(ALTERNATIVE_PIN)
     await sleep(500)
+
+    // Enter an invalid pin and check that we get the correct error and start over
+    await enterPinUi('902100')
+    await expect(element(by.text("The PINs didn't match"))).toBeVisible()
```
Comment (silasbw) [inline e2e/src/usecases/PINChange.js:30]: What in Linter's name?!  (the double quotes)

**Review by silasbw — APPROVED** [review]

File: e2e/src/usecases/PINChange.js (hunk 1 above)
Comment (MuckT) [reply e2e/src/usecases/PINChange.js:30]: Can change to `'The PINs didn\'t match'`.

Comment (silasbw) [reply e2e/src/usecases/PINChange.js:30]: ohhh, i missed that.  sorry.  looks fine.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above issue on Latest iOS Test Flight Release Build V1.36.0 & updated status in test rails.

**Observations:**
- Valid error message “The PIN’s didn’t match” appear when user enter invalid PIN again on “Enter your PIN again to confirm” screen .

**Test rail link:**
**iOS 15:** https://valoraapp.testrail.io/index.php?/tests/view/50657

**Devices:**  iPhone 13 mini (15.1.1)

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above issue on Latest Android Internal Release Build V1.36.0 & updated status in test rails.

**Observation:**
- Valid error message “The PIN’s didn’t match” appear when user enter invalid PIN again on “Enter your PIN again to confirm” screen .

**Test rail link:**
**Android 11:** https://valoraapp.testrail.io/index.php?/tests/view/51724

**Device:** Samsung Galaxy F41(11.0)

Thanks..!!

**Pull request merged**